cloudfront_utils.py

```diff
@@ -0,0 +1,58 @@
+import dns
+import ipaddress
+import json
+import sys
+
+import requests
+
+
+def get_cloudfront_ip_ranges():
+    cloudfront_ip_ranges_url = 'https://d7uri8nf7uskq.cloudfront.net/tools/list-cloudfront-ips'
+    cloudfront_backup_ip_ranges_url = 'https://ip-ranges.amazonaws.com/ip-ranges.json'
+    ip_ranges = []
+
+    try:
+        print('[*] Retrieving Cloudfront IP ranges from {}'.format(cloudfront_ip_ranges_url))
+        page_content = requests.get(cloudfront_ip_ranges_url, timeout=10)
+        ip_ranges_json = json.loads(page_content.text)
+        ip_ranges = sorted({x for v in ip_ranges_json.values() for x in v})
+
+    except requests.exceptions.RequestException:
+        sys.stderr.write('[-] Failed to retrieve Cloudfront IP ranges from cloudfront - trying alternate source')
+
+        try:
+            print('[*] Retrieving Cloudfront IP ranges from {}'.format(cloudfront_backup_ip_ranges_url))
+            page_content = requests.get(cloudfront_backup_ip_ranges_url, timeout=10)
+            ip_ranges_json = json.loads(page_content.text)
+            filtered_v4_ip_ranges = [x['ip_prefix'] for x in ip_ranges_json['prefixes'] if x['service'] == 'CLOUDFRONT']
+            filtered_v6_ip_ranges = [x['ipv6_prefix'] for x in ip_ranges_json['ipv6_prefixes'] if x['service'] == 'CLOUDFRONT']
+            ip_ranges = filtered_v4_ip_ranges + filtered_v6_ip_ranges
+        except requests.exceptions.RequestException:
+            sys.stderr.write('[-] Failed to retrieve Cloudfront IP ranges')
```

Comment:
I didn't add a set of fallback IPs as I don't have a strong conception of how reliable they'd be for CloudFront - and attempt two separate retrieval paths.